src/services/redisSessionStorage.ts

```diff
@@ -125,26 +195,115 @@ export default class RedisSessionStorage implements ISessionStorage {
       );
     });
 
-    const [
-      deleteSessionTokenResult,
-      deleteWalletTokenResult
-    ] = await Promise.all([deleteSessionToken, deleteWalletToken]);
+    const deletePromises = await Promise.all([
+      deleteSessionTokens,
+      deleteWalletToken
+    ]);
 
-    if (isLeft(deleteSessionTokenResult) || isLeft(deleteWalletTokenResult)) {
+    if (
+      deletePromises.some(_ => {
+        return isLeft(_) || !_.value;
+      })
+    ) {
       return left<Error, boolean>(new Error("Error deleting the token"));
     }
+    return right<Error, boolean>(true);
+  }
 
-    if (!deleteSessionTokenResult.value || !deleteWalletTokenResult.value) {
-      return left<Error, boolean>(new Error("Error deleting the token"));
+  public async listUserSessions(
+    user: User
+  ): Promise<Either<Error, SessionsList>> {
+    const keys = await new Promise<Either<Error, ReadonlyArray<string>>>(
+      resolve => {
+        this.redisClient.smembers(
+          `${userSessionsSetKeyPrefix}${user.fiscal_code}`,
+          (err, response) => resolve(this.arrayStringReply(err, response))
+        );
+      }
+    );
+    if (isLeft(keys)) {
+      return left(keys.value);
     }
+    const userSessionTokens = keys.value.map(key => {
+      return new Promise<string>((resolve, reject) => {
+        this.redisClient.get(key, (err, response) => {
+          if (err) {
+            reject(err);
+          }
+          resolve(response);
+        });
+      });
+    });
+    const userSessionTokensResult = await Promise.all(userSessionTokens);
```

Comment:
here you make a call to redis GET for each key, instead you can make a single call to the get value of multiple keys

Reply:
fixed in 708f7a3